### PR TITLE
Feature/sie 59 60 user register login

### PR DESCRIPTION
**Pull request by bleachalon**
Re-created user services with authentication.

1. Created functions for signIn, signUp, getCurrentUser, deleteUser, postUser, and sendEmailVerification.
2. Except emailVerification and signUp, everything else works fine.
3. FE can call postUser for register as for now which register user without email verification.
4. I realize the email verification gets kind of complicated which I will write another ticket for this feature.
5. postUser and signIn should be enough for the prototype.

Let me know if you have any advice for the code.

### CODE REVIEW

**Comment by quanganhhoang** [comment]
how do I test this? Can I pull these changes and try to sign up/log in?

**Comment by bleachalon** [comment]
> how do I test this? Can I pull these changes and try to sign up/log in? EDIT: nvm this will be used w/ #34 right?

Yes, This will be used with #34. If you want to test this separately you need to pull this branch and create js functions to call these services yourself.

**Comment by alecherryy** [comment]
> how do I test this? Can I pull these changes and try to sign up/log in? EDIT: nvm this will be used w/ #34 right?

I merged this branch into mine and tested the response, it all seems to work as expected!

File: src/firebase/CRUDServices/userServices.js [hunk 1]
```diff
@@ -21,97 +17,120 @@ export default class UserServices {
...
     try {
-      const userRef = db.collection(self.users)
-        .doc(userId);
-      await userRef.update(user);
-      return true;
-    } catch (err) {
-      // TODO: add logic here
-      return false;
+      const userCredential = await firebase.auth()
+        .signInWithEmailAndPassword(user.email, user.password);
```
Comment (alecherryy) [inline src/firebase/CRUDServices/userServices.js:85]: I thought we were doing usernames. Is it email instead?

**Comment by alecherryy** [comment]
> > how do I test this? Can I pull these changes and try to sign up/log in? EDIT: nvm this will be used w/ #34 right?
> 
> Yes, This will be used with #34. If you want to test this separately you need to pull this branch and create js functions to call these services yourself.

I merged your branch into mine, updated the code and keep getting this ErrorMessage `"There is no user record corresponding to this identifier. The user may have been deleted."` even tho I am using my sample credentials which can be found on Firebase, under Users. Email: `pizzoccheri.a@northeastern.edu`, password: `1234`

File: src/firebase/CRUDServices/userServices.js (hunk 1 above)
Comment (bleachalon) [reply src/firebase/CRUDServices/userServices.js:85]: Yes, sorry for the confusion. Firestore seems like not supporting username as standard login.

**Comment by bleachalon** [comment]
> > > how do I test this? Can I pull these changes and try to sign up/log in? EDIT: nvm this will be used w/ #34 right?
> > 
> > 
> > Yes, This will be used with #34. If you want to test this separately you need to pull this branch and create js functions to call these services yourself.
> 
> I merged your branch into mine, updated the code and keep getting this ErrorMessage `"There is no user record corresponding to this identifier. The user may have been deleted."` even tho I am using my sample credentials which can be found on Firebase, under Users. Username: `pizzoccheri.a`. I have been using the email and password.

I need more details to identify the issue you mentioned. Do you have time to meet and show me what you are experiencing?

**Comment by alecherryy** [comment]
> > > > how do I test this? Can I pull these changes and try to sign up/log in? EDIT: nvm this will be used w/ #34 right?
> > > 
> > > 
> > > Yes, This will be used with #34. If you want to test this separately you need to pull this branch and create js functions to call these services yourself.
> > 
> > 
> > I merged your branch into mine, updated the code and keep getting this ErrorMessage `"There is no user record corresponding to this identifier. The user may have been deleted."` even tho I am using my sample credentials which can be found on Firebase, under Users. Username: `pizzoccheri.a`. I have been using the email and password.
> 
> I need more details to identify the issue you mentioned. Do you have time to meet and show me what you are experiencing?

Not sure what's different, but all works now!

**Review by alecherryy — APPROVED** [review]

**Comment by alecherryy** [comment]
@quanganhhoang can you take a look at this? I'd like to merge it if it's good. I tested it on my local and it worked.

**Comment by quanganhhoang** [comment]
> @quanganhhoang can you take a look at this? I'd like to merge it if it's good. I tested it on my local and it worked.

SGTM. Let's just check it in and note to ourselves to clean up later if necessary

**Review by quanganhhoang — APPROVED** [review]

**Pull request merged**